### PR TITLE
Parameterize Civ Unique for increased XP gain

### PR DESCRIPTION
**Pull request by SpacedOutChicken**
This change allows for individual civilizations and mods to increase nationwide XP gain at a rate of the modder's choosing, or even to create a nation that gains XP more slowly than everyone else.

### CODE REVIEW

File: core/src/com/unciv/logic/battle/Battle.kt [hunk 1]
```diff
@@ -258,7 +258,10 @@ object Battle {
             return
 
         var XPModifier = 1f
+        // keep until deprecated
```
Comment (yairm210) [inline core/src/com/unciv/logic/battle/Battle.kt:261]: Please add what version this was added into so we'll know how much time has passed :)

File: core/src/com/unciv/logic/battle/Battle.kt [hunk 2]
```diff
@@ -258,7 +258,10 @@ object Battle {
...
         if (thisCombatant.getCivInfo().hasUnique("Military units gain 50% more Experience from combat")) XPModifier += 0.5f
+        for (unique in thisCombatant.getCivInfo().getMatchingUniques("Military units gain []% more Experience from combat"))
```
Comment (yairm210) [inline core/src/com/unciv/logic/battle/Battle.kt:263]: If you're already parametrizing this, we can turn the Mitilary units into [] units, to allow specific unit types to gain more exp for specific civs or from buildings

Comment (SpacedOutChicken) [reply core/src/com/unciv/logic/battle/Battle.kt:263]: Understood. Should I close this pull request and open a new one?

**Comment by SpacedOutChicken** [comment]
I tried adding more parameters and it broke somewhere. Not sure what went wrong.

File: core/src/com/unciv/logic/battle/Battle.kt [hunk 3]
```diff
@@ -258,7 +258,12 @@ object Battle {
             return
 
         var XPModifier = 1f
+        // deprecated as of 3.12.10, keeping this for now in to avoid breaking mods
         if (thisCombatant.getCivInfo().hasUnique("Military units gain 50% more Experience from combat")) XPModifier += 0.5f
+        for (unique in thisCombatant.getCivInfo().getMatchingUniques("[] units gain []% more Experience from combat")) {
+            if (unit.matchesFilter(unique.params[0]))
```
Comment (yairm210) [inline core/src/com/unciv/logic/battle/Battle.kt:264]: thisCombatant.unit  instea of just unit :)

**Pull request merged**